src/dependency-container.ts

```diff
@@ -360,6 +441,36 @@ class InternalDependencyContainer implements DependencyContainer {
     return childContainer;
   }
 
+  beforeResolution<T>(
+    token: InjectionToken<T>,
+    callback: PreResolutionInterceptorCallback<T>,
+    options?: InterceptorOptions
+  ): void {
+    if (options == null) {
+      options = {frequency: Frequency.Always};
+    }
```

Comment:
You can move this to be a default param value

Reply:
I tried this, and I get an error that a parameter initializer is only allowed on the constructor. That's kinda weird.